core/src/main/java/com/segment/analytics/kotlin/core/utilities/DateTimeUtils.kt

```diff
@@ -12,8 +12,10 @@ import java.util.*
  * dateTimeNowString(): 2023-04-19T04:03:46.880Z
  */
 fun dateTimeNowString(): String {
-    val sdf = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'.'Szzz")
+    // Note, we should specify locale = Locale.ROOT, otherwise the timestamp returned will use
+    // the default locale, which may not be what we want.
+    val sdf = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'.'Szzz", Locale.ROOT)
```

Comment:
The [Android docs](https://developer.android.com/reference/java/util/Locale) recommend using `Locale.US`, but since this might be used in JVM projects I opted to go with `Locale.ROOT`

> The default locale is not appropriate for machine-readable output. The best choice there is usually Locale.US – this locale is guaranteed to be available on all devices, and the fact that it has no surprising special cases and is frequently used (especially for computer-computer communication) means that it tends to be the most efficient choice too.